packages/kit/src/internal/esm.ts

```diff
@@ -2,18 +2,30 @@ import { fileURLToPath, pathToFileURL } from 'node:url'
 import { interopDefault, resolvePath, resolvePathSync } from 'mlly'
 import { createJiti } from 'jiti'
 import { captureStackTrace } from 'errx'
+import { join } from 'pathe'
 
 export interface ResolveModuleOptions {
+  /** @deprecated use `url` with URLs pointing at a file - never a directory */
   paths?: string | string[]
+  url?: URL | URL[]
 }
 
+export function directoryToParentURL (dir: string): URL {
+  return pathToFileURL(join(dir, '_index.js'))
+}
+
+const importMetaURL = new URL(import.meta.url)
```

Comment:
I think this can be simplified too

Reply:
I was sort of imagining `mlly` could short circuit something if it was a URL instance, but if it doesn't matter....

Reply:
it does by checking `startsWith('file://')` (also lots of old logic there, `URL` constructor is overhead most of the times)